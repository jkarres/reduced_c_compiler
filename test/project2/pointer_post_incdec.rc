// testing ptr++ and ptr--

structdef S {
  int[100] ia;
};

S[10] sa;

function : void main() {

  // first putting something in sa[1].ia[0]
  sa[0].ia[0] = 42;

  // now trying to get a pointer pointing there
  S* ps = &sa[0];
  
  // and printing out what we have, incrementing ps in the processing
  cout << ps++->ia[0] << endl;


  // now the same thing with a pointer to int
  int[10] ia;

  ia[0] = 99;
  int * pi = &ia[0];
  cout << *pi++ << endl;



  // and now doing the same thing to the [0] elements
  ps--;
  pi--;

  ps->ia[0] = -42;
  *pi = -99;

  cout << sa[0].ia[0] << endl;
  cout << ia[0] << endl;

}
