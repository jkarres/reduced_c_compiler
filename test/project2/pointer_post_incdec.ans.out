42
99
-42
-99
